### PR TITLE
feat: SC: 35536 - Add optional xss for padding on accordion

### PR DESCRIPTION
**Pull request by kbesingeryeh**
# [SC-35536](https://app.shortcut.com/homebound-team/story/35536/add-omitpadding-to-accordionlist-in-beam-and-update-financials-widget)
* Adds optional `xss` prop in order to have control over changing the padding on the blueprint side

### CODE REVIEW

File: src/components/Accordion.tsx [hunk 1]
```diff
@@ -23,6 +23,8 @@ export interface AccordionProps {
   setExpandedIndex?: Dispatch<SetStateAction<number | undefined>>;
   /** Used by Accordion list. Sets default padding to 0 for nested accordions */
   omitPadding?: boolean;
+  /** Styles overrides for omit padding */
+  xss?: Xss<Padding>;
 }
 
 export function Accordion(props: AccordionProps) {
```
Comment (bdow) [inline src/components/Accordion.tsx:30]: **request** In order for the XSS to property to only allow `Padding`, then we need to add some more to this component's function definition.
```
type AccordionXss = Xss<Padding>

export interface AccordionProps<X> {
  xss?: X;
}

export function Accordion<X extends Only<AccordionXss, X>>(props: AccordionProps<X>) {
```

File: src/components/Accordion.tsx [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { useId, useResizeObserver } from "@react-aria/utils";
...
 import { useFocusRing } from "react-aria";
 import { Icon } from "src/components/Icon";
-import { Css, Palette } from "src/Css";
+import { Css, Only, Padding, Palette, Xss } from "src/Css";
 import { useTestIds } from "src/utils";
 
-export interface AccordionProps {
+type AccordionXss = Xss<Padding>;
+
+export interface AccordionProps<X> {
```
Comment (stephenh) [inline src/components/Accordion.tsx:10]: Suggestion: I think the build can be fixed by making this:

```
export interface AccordingProps<X = AccordionXss> {
```

**Review by stephenh — APPROVED** [review]
Cool!

File: src/components/Accordion.tsx (hunk 2 above)
Comment (kbesingeryeh) [reply src/components/Accordion.tsx:10]: got it. ty!

**Pull request merged**